git-edit-index

```diff
@@ -194,12 +194,14 @@ def edit_index(index):
     """
     # We need to use a temporary file to store the current index and show it to
     # the user in an editor.
```

Comment:
Here, I would add a comment describing the reason behind the implementation (e.g. why do we need `os.fdopen()`). This is pretty much what you have written in the PR.